src/main/java/com/ironcorelabs/tenantsecurity/kms/v1/DocumentMetadata.java

```diff
@@ -10,6 +10,10 @@
  * classification, as well as optional fields for other arbitrary key/value pairs and a request ID
  * to send to the Tenant Security Proxy.
  */
+// TODO: is it worth making another one of these (exact same) called FieldMetadata for
```

Comment:
We could just move all the code to a "base" class and then make both extend it. Might be worth it for clarity.

Reply:
I made a `Metadata` class and `DocumentMetadata` and `FieldMetadata` extend it. Please take a look to make sure I did it right.

Reply:
Checked it, looks good.